src/multirw.rs

```diff
@@ -0,0 +1,256 @@
+{
+    pub(crate) fn new(txn: RoTransaction) -> MultiReader<K> {
+        MultiReader {
+            tx: txn,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Provides a cursor to all of the values for the duplicate entries that match this key
+    pub fn get(&self, store: MultiStore, k: K) -> Result<Iter, StoreError> {
+        let mut cursor = self.tx.open_ro_cursor(store.0).map_err(StoreError::LmdbError)?;
+        let iter = cursor.iter_dup_of(k);
+        //Ok(MultiIter { iter, cursor, })
+        Ok(Iter {
+            iter,
+            cursor,
+        })
+    }
+
+    /// Provides a cursor to all of the values for the duplicate entries that match this key
+    pub fn get_first(&self, store: MultiStore, k: K) -> Result<Option<OwnedValue>, StoreError> {
+        let mut cursor = self.tx.open_ro_cursor(store.0).map_err(StoreError::LmdbError)?;
+        let mut iter = cursor.iter_dup_of(k);
+        read_transform_owned(iter.next().ok_or(lmdb::Error::NotFound).map(|e| e.1))
+    }
+
+    /// Cancel this read transaction
+    pub fn abort(self) {
+        self.tx.abort();
+    }
+
+    /* TODO - Figure out how to solve the need to have the cursor stick around when 
+     *        we are producing iterators from MultiIter 
+    /// Provides an iterator starting at the lexographically smallest value in the store
+    pub fn iter_start(&self, store: MultiStore) -> Result<MultiIter, StoreError> {
+        let mut cursor = self.tx.open_ro_cursor(store.0).map_err(StoreError::LmdbError)?;
+
+        // We call Cursor.iter() instead of Cursor.iter_start() because
+        // the latter panics at "called `Result::unwrap()` on an `Err` value:
+        // NotFound" when there are no items in the store, whereas the former
+        // returns an iterator that yields no items.
+        //
+        // And since we create the Cursor and don't change its position, we can
+        // be sure that a call to Cursor.iter() will start at the beginning.
+        //
+        let iter = cursor.iter_dup();
+
+        Ok(MultiIter {
+            iter,
+            cursor,
+        })
+    }
+    */
+}
+
+/*
+impl<'env> Iterator for MultiIter<'env> {
+    type Item = Iter<'env>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.iter.next() {
+            None => None,
+            Some(iter) => Some(Iter {
+                iter,
+                cursor,
+            }),
+        }
+    }
+}
+*/
+
+impl<'env> Iterator for Iter<'env> {
+    type Item = (&'env [u8], Result<Option<Value<'env>>, StoreError>);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.iter.next() {
+            None => None,
+            Some((key, bytes)) => Some((key, read_transform(Ok(bytes)))),
+        }
+    }
+}
+
+/// New type around an `lmdb::Database`.  At this time, the underlying LMDB
+/// handle (within lmdb-rs::Database) is a C integer, so Copy is automatic.
+#[derive(Copy, Clone)]
+pub struct MultiStore(Database);
```

Comment:
Should we make the *Database* inside the *Store* struct be `pub(crate)` and have *MultiStore* wrap a *Store*, like *IntegerStore* does?